### DISCUSSION TITLE
How to train DreamBooth with a set of prompts and images

### DISCUSSION THREAD

**Discussion by xiong224** [open-mmlab/mmediting#1850] [Models]
Thanks for your work. I find that the mmagic can only train DreamBooth with a fixed prompt and how can I train DreamBooth with a set of prompts and images? If there is anything I‘ve missed, thanks for reminding me.

**Comment by LeoXing1996** [comment]
Hey @xiong224, you are right. The current version of MMagic does not support DreamBooth with multi prompts.

You can modify the training code as follow:
```diff
input_ids = self.tokenizer(
-   prompt,
+   random.sample(self.prompts, latents.shape[0]),
    max_length=self.tokenizer.model_max_length,
    return_tensors='pt',
    padding='max_length',
    truncation=True)['input_ids'].to(self.device)

```
